.travis.yml

```diff
@@ -7,13 +7,16 @@ services:
 matrix:
   include:
     - env: TO_TEST=build
-    - env: TO_TEST=e2e
+    - env: TO_TEST=ephemeral
+    - env: TO_TEST=pyspark-templates
+    - env: TO_TEST=java-templates
+    - env: TO_TEST=scala-templates
   fast_finish: true
 
 before_install:
 ## add insecure-registry and restart docker
 - |
-  if [ "$TO_TEST" = "e2e" ]; then
+  if [ "$TO_TEST" != "build" ]; then
```

Comment:
ack, we could do that.  It makes the script harder to read, but easier to debug I agree